### PR TITLE
refactor: move example packages to examples folder

### LINKED ISSUE #1218: Move packages/examples to main folder
## Description / Steps to reproduce / Feature proposal
Issue #836 brought the examples back into the monorepo (_/packages/examples_). However, the **packages** folder is already substantially sized. A suggestion is to make a dedicated examples folder in the main project folder, as these have nothing to do with the actual LoopBack REST API logic in the **packages** folder, but just show how to get started or work with LoopBack 4.

## Current Behavior
The example folders (example-getting-started, example-hello-world, example-log-extension, example-rpc-server) are in the packages folder.

## Expected Behavior
The example folders are in the main folder and the packages folder is smaller and more organized.

### PR DESCRIPTION
**Pull request by jwooning**
The packages folder currently contains all the examples, the example packages don't contain any
loopback logic, therefore it makes sense to move the examples to a separate folder. All example
packages are thus moved from `packages/example-{name}` to `examples/{name}`.

Note that not all tests succeed, because the cli examples generator tries to fetch the examples from the changed examples structure, which will fail because the master is not yet restructured.

Fix #1218 

## Checklist

- [ ] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [x] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,8 +5,10 @@ npm-debug.log
 coverage
 .nyc_output
 api-docs
-packages/*/*.tgz
+**/*.tgz
 packages/*/dist*
 packages/*/package
 packages/_sandbox
 .sandbox
+examples/*/dist*
```
Comment (shimks) [inline .gitignore]: Let's be consistent and use `**/dist*` as well

File: .prettierignore [hunk 2]
```diff
@@ -1,5 +1,5 @@
-packages/*/dist
 packages/*/api-docs
 packages/cli/generators/*/templates
+**/*/dist
```
Comment (virkt25) [inline .prettierignore]: I think just a `**/dist` should work.

**Comment by virkt25** [comment]
Merging as it's a trivial PR but a big one and don't want to have it require another rebase.

**Comment by virkt25** [comment]
@jwooning Thanks for your contribution!! Your PR has landed 🎉

**Pull request merged**